addOns/bruteforce/src/main/java/org/zaproxy/zap/extension/bruteforce/ExtensionBruteForce.java

```diff
@@ -152,16 +168,158 @@ public void optionsLoaded() {
         }
     }
 
-    protected void bruteForceSite(SiteNode siteNode) {
-        this.getBruteForcePanel().bruteForceSite(siteNode);
+    public int bruteForceSite(SiteNode siteNode, String payloadFileName) {
+        for (ForcedBrowseFile file : this.getFileList()) {
+            if (file.getFile().getName().equals(payloadFileName)) {
+                return this.bruteForceSite(siteNode, file.getFile());
+            }
+        }
+        throw new IllegalArgumentException("File not found: " + payloadFileName);
+    }
+
+    public int bruteForceSite(SiteNode siteNode, File payloadFile) {
+        ScanTarget scanTarget = createScanTarget(siteNode);
```

Comment:
It should validate that the `siteNode` is not null otherwise it will fail later (and seems to break the panel).